app/repositories/CacheRepository.scala

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package repositories
+
+import config.AppConfig
+import models.{Frontend, UserAnswers}
+import org.mongodb.scala.model.Indexes.{ascending, compoundIndex}
+import org.mongodb.scala.model._
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+
+import java.time.LocalDateTime
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+class CacheRepository(
+  mongoComponent: MongoComponent,
+  appConfig: AppConfig,
+  frontend: Frontend
+)(implicit ec: ExecutionContext)
+    extends PlayMongoRepository[UserAnswers](
+      mongoComponent = mongoComponent,
+      collectionName = frontend.collectionName,
+      domainFormat = UserAnswers.format,
+      indexes = CacheRepository.indexes(appConfig)
+    ) {
+
+  def get(lrn: String, eoriNumber: String): Future[Option[UserAnswers]] = {
+    val filter = Filters.and(
+      Filters.eq("lrn", lrn),
+      Filters.eq("eoriNumber", eoriNumber)
+    )
+    val update = Updates.set("lastUpdated", LocalDateTime.now())
+
+    collection
+      .findOneAndUpdate(filter, update, FindOneAndUpdateOptions().upsert(false))
```

Comment:
Why are we doing an update on a get? Is this to prevent cache expiry if a user is just dipping in and out with no updates?

Reply:
Yeah I believe this is a standard pattern to update the `lastUpdated` value for both a `get` and a `set`. Interestingly the code that comes pre-built when creating a microservice with mongo does it across two calls (https://github.com/hmrc/manage-transit-movements-prelodge-frontend/blob/d6568f8a743f3f244a45d4a03aa5aa77894c3bee/app/repositories/SessionRepository.scala#L51-L68)